### PR TITLE
created an MVC for getting review info by ID

### PR DESCRIPTION
**Pull request by JamesATeasdale**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -32,3 +32,37 @@ describe("200 GET /api/categories should return all categories with the followin
...
+			});
+	});
+	test("should return a 404 error with a corresponding message when no review is found", () => {
+		return request(app)
+			.get("/api/reviews/908")
+			.expect(404)
+			.then(({ body }) => {
+				expect(body).toEqual({ msg: "Not found" });
+			});
+	});
```
Comment (B7Bprog) [inline __tests__/app.test.js:66]: Please include a test for a 400, bad request, when the id is the wrong data type, for example a string. (/api/reviews/banana). You can check the psql error code for that and reject the promise based on that one.

**Review by B7Bprog — CHANGES_REQUESTED** [review]
Please read my comment and add the missing test. After it's done, please send us a quick pull request with it.

File: controllers/app.controller.js [hunk 2]
```diff
@@ -1,6 +1,18 @@
...
+exports.getCategories = async (req, res) => {
 	const categories = await fetchCategories();
 	res.status(200).send({ categories: categories.rows });
 };
+
+exports.getReview = async (req, res, next) => {
+	const reviewId = req.params["review_id"];
+  try {
+		const review = await fetchReview(reviewId);
+		review.rowCount === 0
```
Comment (callumu21) [inline controllers/app.controller.js]: This logic should be handled in the model when we choose whether to return the data or a rejected Promise.

File: app.js [hunk 3]
```diff
@@ -1,12 +1,17 @@
...
 
 app.use(express.json());
 
 app.get("/api/categories", getCategories);
 
+app.get("/api/reviews/:review_id", getReview);
+
 app.all("/*", notFound);
 
+app.use("/", badDataType);
```
Comment (callumu21) [inline app.js]: app.use just needs an error-handling middleware function as its argument, not a path as well.

**Review by callumu21 — APPROVED** [review]
Nice to see some newer syntax in use here (async, try/catch, ternary operators)! I've left some comments that should be addressed before you merge:

1. Amend app.use to not take a path as its initial argument
2. You should handle the rejection of your db.query in the model when there are no reviews, rather than the controller.

**Review by HeatherEver — APPROVED** [review]
Nice work here! The only thing i would say is that i see you've used `async...await` in one controller and `.then promises` in another - i would say stick to one or the other in the same project just for consistency purposes

**Pull request merged**